### ISSUE TITLE
Angular velocity also needs to be tracked

### ISSUE THREAD

**Issue by haje-hwang** [SitronX/UnityTimeRewinder#4]
when i tryed your project, I found someting weird.

![spin_get_faster](https://user-images.githubusercontent.com/80046755/230765480-612cc971-96b6-46d4-89f7-8ee5dbd3f827.gif)
Every time I rewind the time, it seemed that the cube is spinning faster and faster.

![more_significant](https://user-images.githubusercontent.com/80046755/230765483-3782588a-e631-4529-8e61-3990d54b623b.gif)
I tested it further by adding another object. When I rotated the times, the objects did not return to their original angle.

Later I found that this project does not track angular velocity. I think that was the cause.

So I changed it to track the angular velocity as well, mimicking CircularBuffer<Vector3> trackedVelocities in RewindAbstracts.cs
 this is what i add in RewindAbstracts.cs, Line 116
 
` #region AngularVelocity
    CircularBuffer<Vector3> trackedAngularVelocities;
    /// <summary>
    /// Call this method in Track() if you want to track angular velocity of Rigidbody
    /// </summary>
    protected void TrackAngularVelocity()
    {

        if (body != null)
        {
            trackedAngularVelocities.WriteLastValue(body.angularVelocity);            
        }
        else if (body2!=null)
        {
            trackedAngularVelocities.WriteLastValue(new Vector3(0, 0, body2.angularVelocity));
        }
        else
        {
            Debug.LogError("Cannot find Rigidbody on the object, while TrackAngularVelocity() is being called!!!");
        }
    }
    /// <summary>
    /// Call this method in GetSnapshotFromSavedValues() to angular velocity of Rigidbody
    /// </summary>
    protected void RestoreAngularVelocity(float seconds)
    {   
        if(body!=null)
        {
            body.angularVelocity = trackedVelocities.ReadFromBuffer(seconds);
        }
        else
        {
            body2.angularVelocity = trackedVelocities.ReadFromBuffer(seconds).z;
        }
    }
    #endregion`
    

 and applied this in GenericRewind.cs
`public class GenericRewind : RewindAbstract
{
    [SerializeField] bool trackPositionRotation;
    [SerializeField] bool trackVelocity;
    [SerializeField] bool trackAngularVelocity;
    [SerializeField] bool trackAnimator;
    [SerializeField] bool trackAudio;
    [SerializeField] bool trackParticles;

    [Tooltip("Fill particle settings only if you check Track Particles")]
    [SerializeField] ParticlesSetting particleSettings;

    protected override void Rewind(float seconds)
    {

        if (trackPositionRotation)
            RestorePositionAndRotation(seconds);
        if (trackVelocity)
            RestoreVelocity(seconds);
         if (trackAngularVelocity)
            RestoreAngularVelocity(seconds);
        if (trackAnimator)
            RestoreAnimator(seconds);
        if (trackParticles)
            RestoreParticles(seconds);
        if(trackAudio)
            RestoreAudio(seconds);
    }

    protected override void Track()
    {
        if (trackPositionRotation)
            TrackPositionAndRotation();
        if (trackVelocity)
            TrackVelocity();
        if (trackAngularVelocity)
            TrackAngularVelocity();
        if (trackAnimator)
            TrackAnimator();
        if (trackParticles)
            TrackParticles();
        if (trackAudio)
            TrackAudio();
    }
    private void Start()
    {
        InitializeParticles(particleSettings);
    }

}`

**Comment by SitronX** [comment]
Hello @haje-hwang ,
I actually know about this issue. You are right, the angular velocity also needs to be tracked.

I have started implementing fix for this with few other changes while ago, but sadly i had to put it down for a while and then i completely forgot about it.

Your solution works well, but it would probably be better to track angular velocity by default, so if the user uses his custom rewind script, he doesnt need to call TrackVelocity() and TrackAngularVelocity(), just only TrackVelocity() for simplification.

Thank you for this reminder, i will upload the fix hopefully today when i finish testing all the changes i have done. Sorry for the trouble, it could have been fixed already with new version, but i simply forgot.
